### PR TITLE
feat: Add CHAIN_ID_INDICES to config store

### PR DESCRIPTION
**Pull request by nicholaspai**
Long standing change to move CHAIN_ID list to config store rather than enshrined only in UMIP

### CODE REVIEW

File: UMIPs/umip-157.md [hunk 1]
```diff
@@ -93,14 +93,13 @@ The following constants should reflect what is stored in the [`AcrossConfigStore
 - "VERSION"
   - Across protocol version number. Supporting implementations should query this value against the value defined in their implementation to determine compatibility with the current protocol version. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of funds. For more information go [here](#versions).
 - "DISABLED_CHAINS"
-  - This must be a stringified list of chain ID numbers. This cannot contain the chain ID "1".
+  - This must be a stringified list of chain ID numbers. This cannot contain the chain ID "1". Chains in here must be contained in `CHAIN_ID_INDICES`.
```
Comment (pxrl) [inline UMIPs/umip-157.md:96]: `This cannot contain the chain ID "1".`

1, or the HubPool chain ID?

File: UMIPs/umip-157.md [hunk 2]
```diff
@@ -93,14 +93,13 @@ The following constants should reflect what is stored in the [`AcrossConfigStore
...
+- "CHAIN_ID_INDICES"
+  - This should default to the value [1,10,137,288,42161]. Chains can only be added to this list to be valid.
```
Comment (pxrl) [inline UMIPs/umip-157.md:98]: Do we want to hardcode the default chain IDs in the UMIP? It'd mean that we need to update the UMIP to append a new chain ID each time. Given that we're referencing ConfigStore config, it seems redundant to specify a default array here.

Comment (nicholaspai) [reply UMIPs/umip-157.md:98]: i am just hard coding these to make up for the fact that all blocks prior to now won't have set this variable. Going forward we can always use this variable set in the config store

File: UMIPs/umip-157.md (hunk 1 above)
Comment (nicholaspai) [reply UMIPs/umip-157.md:96]: good clarification will add

File: UMIPs/umip-157.md (hunk 2 above)
Comment (nicholaspai) [reply UMIPs/umip-157.md]: 7a3d2856f5d2ebd314e2168a9ecf3116285f225d

**Review by mrice32 — APPROVED** [review]

**Review by james-a-morris — APPROVED** [review]
This is a good change.

**Pull request merged**